packages/app-cli/tests/services/plugins/api/JoplinWorkspace.ts

```diff
@@ -81,4 +82,31 @@ describe('JoplinWorkspace', () => {
 		expect(modFolder.title).toBe('changedtitle');
 	});
 
+	test('should remove event listeners when plugins are unloaded', async () => {
+		const service = newPluginService();
+
+		const pluginScript = newPluginScript(`
+			joplin.plugins.register({
+				onStart: async () => {
+					for (let i = 0; i < 8; i++) {
+						await joplin.workspace.onNoteChange(async (event) => { });
+						await joplin.workspace.onResourceChange(async (event) => { });
+						await joplin.workspace.filterEditorContextMenu(async (event) => { });
+					}
+				}
+			})
+		`);
+		const plugin = await service.loadPluginFromJsBundle('', pluginScript);
+		await service.runPlugin(plugin);
+
+		const itemChangeListenerCounter = eventManager.listenerCounter_(EventName.ItemChange);
+		const resourceChangeListenerCounter = eventManager.listenerCounter_(EventName.ResourceChange);
+
+		plugin.onUnload();
```

Comment:
Currently, this test uses `plugin.onUnload` rather than `service.unloadPlugin`. This is because, at present, neither `loadPluginFromJsBundle` nor `runPlugin` adds the plugin to `PluginService.plugins_` (this seems to only be done by `PluginService.loadAndRunPlugins`).

It may make sense to change this, perhaps by also adding plugins to `PluginService.plugins_` in `.runPlugin`. As I'm unsure why plugins are currently added only in `.loadAndRunPlugins`, I would prefer to make this change in a separate pull request so that it's easier to test and revert if it causes a regression.